### DISCUSSION TITLE
How to handle big file size uploads (potentially more than 5Gb) using Supabase Storage ?

### DISCUSSION THREAD

**Discussion by TeddyLourson** [supabase/supabase#6029] [Questions]
Hi 👋 !
I'm building a marketplace for digital artists and I was wondering how could I allow users to upload big files ?
People would be able to sell 3D models with potentially huge image textures (could be 4K or even more), multiple resolutions for their 3D models (LODs), multi-resolution modifiers and all of those things may add up to the file size.
My first thought would be to compress the files into a zip, split it into small parts and upload those to a specific directory, then merge them back when a user downloads it but I want something easier for the user to work with. I don't want to ask them to open the terminal and run a command to split a zip file into multiple part and the same for buyers who probably expect to get a single zip file.
Since there's a limit of 50mb for free users and 5Gb for premium users using Supabase Storage, what could be my options ?

Thanks in advance !

**Comment by TeddyLourson** [comment]
I found this node package ([node-split-file](https://github.com/tomvlk/node-split-file)) which could actually help me implement the technique I mentioned in my question above.
I'll try to cook something up with it and come back share my solution so that someone could get some inspiration from it in the future (if that even works).

**Reply by fadethegap** [reply]
@Teio07 Have you come up with a solution to this? I submitted a support ticket regarding the same issue.

**Reply by TeddyLourson** [reply]
Not yet, I’ll be using files under 50Mb for now as I’m not in production right now and work on other features. Since I’m also on vacation, I didn’t work on my website for a week, but I might give it a try in something like a month or so depending on how fast I can finish other features and I’ll share my solution if I find any.
